src/interfaces/libpq/libpq-int.h

```diff
@@ -83,6 +83,34 @@ typedef struct
 #endif
 #endif   /* USE_SSL */
 
+/*
+ * Definitions meant to lessen the malloc() usage for CDB routines in
+ * server-sided code for struct PGresult allocations.
+ */
+#ifndef FRONTEND
+#include "postgres.h"
+#include "utils/memutils.h"
+
+/*
+ * CurTransactionContext's lifetime lasts until the end of the current query,
+ * which does the job well for backends. Auxillary processes do not set
+ * transaction contexts, so we have to use TopMemoryContext to achieve a
+ * lifetime equivalent to malloc().
+ */
+#define PQ_PALLOC_CONTEXT \
+	((CurTransactionContext != NULL) ? CurTransactionContext : TopMemoryContext)
+
+#define maybe_palloc(sz) MemoryContextAlloc(PQ_PALLOC_CONTEXT, sz)
+#define maybe_pstrdup(x) MemoryContextStrdup(PQ_PALLOC_CONTEXT, x)
+#define maybe_repalloc(x, sz) repalloc(x, sz)
+#define maybe_pfree(x) pfree(x)
+#else
+#define maybe_palloc(sz) malloc(sz)
+#define maybe_pstrdup(x) strdup(x)
+#define maybe_repalloc(x, sz) realloc(x, sz)
+#define maybe_pfree(x) free(x)
```

Comment:
I don't really like the current names, how about these?
```suggestion
#define be_fe_alloc(sz) MemoryContextAlloc(PQ_PALLOC_CONTEXT, sz)
#define be_fe_strdup(x) MemoryContextStrdup(PQ_PALLOC_CONTEXT, x)
#define be_fe_realloc(x, sz) repalloc(x, sz)
#define be_fe_free(x) pfree(x)
#else
#define be_fe_alloc(sz) malloc(sz)
#define be_fe_strdup(x) strdup(x)
#define be_fe_realloc(x, sz) realloc(x, sz)
#define be_fe_free(x) free(x)
```

Reply:
I don't think it's a more informative name that the current one still. Maybe leave it as is, or use something like `malloc_or_palloc` naming, if you insist?

Reply:
How about `bf_alloc`? (**b**_ackend_ or **b**_rontend_ **alloc**)

Reply:
I think it's unintuitive too. The main reason we may be doing this is to allow people who read the code to understand what the function is doing without seeing the definition. I think `maybe_...()` or `..._or_...()` names suit this purpose the best.

Reply:
`pg_malloc`?

Reply:
I also think something like `pg_malloc` would be better.

Reply:
There already is `pg_malloc()`.

Reply:
> There already is `pg_malloc()`

`pq_malloc`